include/exadg/matrix_free/categorization.h

```diff
@@ -50,8 +50,12 @@ do_cell_based_loops(dealii::Triangulation<dim> const & tria,
   else
     data.cell_vectorization_category.resize(tria.n_active_cells());
 
+  unsigned int const n_faces_per_cell = tria.get_reference_cells()[0].n_faces();
```

Comment:
The changes in this function look fine. But, I would like to point out that cell-centered face loops do not work yet in deal.II.

Reply:
I assume you mean for simplex? 

Yes, I have noticed that. As far as I remember, the code failed here for me:

https://github.com/dealii/dealii/blob/947199be8142d3fc7bde15d8f94f168573ccf5d2/include/deal.II/grid/tria_accessor.templates.h#L755-L764 

And once you mentioned that this Assert is mostly an indication of the function not being extended, yet.

Reply:
I think we should assert in ExaDG (in categorization.h) that simplices are not supported for cell-based face loops. What do you think?

Principally, we should only rely on deal.II error messages if the error message is easy to interpret for ExaDG users and if this error message appears both in debug and release mode. To make an example, I encountered recently that the degree <=2 restriction of simplices in dealii does not give easy to interpret error messages in release mode, which is clearly sub-optimal for ExaDG users.

Reply:
> I think we should assert in ExaDG (in categorization.h) that simplices are not supported for cell-based face loops. What do you think?

I agree. I had different kind of errors from dealii in Debug mode, when I tried to enable cell-based face loops, see https://github.com/exadg/exadg/pull/380#issuecomment-1464139651. And I would say, they would be hard to interpret for someone who is not familiar with simplices.

However, I am not sure if we should for example assert degree <=2 in ExaDG, as well. Most of the dealii asserts I have encountered only appear in debug mode. I think if we add them on ExaDG, we would end up having too many. That is why I do development in debug, and calculations in release mode.

Reply:
Sure, but let's also keep in mind that you are the person most familiar with simplices in ExaDG. We should make sure that every other user of ExaDG is able to use simplices as well. I think if we place the asserts at a central place, its not too much work. Let's postpone the discussion on the "degree <=2 restriction", it is not related to this PR and was just an example. (The problem with degree <=2 is that we do not construct the finite element at a central place in ExaDG for all the modules, meaning that we would need this assert at several places, which is clearly sub-optimal.)

Reply:
@peterrum what do you suggest here? An additional assert for simplex?